cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -95,11 +96,19 @@ public HBaseMetricsTable(DatasetContext datasetContext, DatasetSpecification spe
     this.columnFamily = TableProperties.getColumnFamilyBytes(spec.getProperties());
   }
 
-  private void initializeV3Vars(DatasetSpecification spec) {
+  private void initializeV3Vars(CConfiguration cConf, DatasetSpecification spec) {
     boolean isV3Table = spec.getName().contains("v3");
     this.scanExecutor = null;
     this.rowKeyDistributor = null;
     if (isV3Table) {
+      // If v3 table is already created, we will ignore splits value from cConf to avoid modifying splits after creation
```

Comment:
This warnings will be logged for each thread having this dataset. Even though it does not flood the master logs on cluster, it does get logged more than 4 times from different threads I believe. 

The problem in adding it in `MessagingMetricsProcessorService` is the dataset spec is not exposed by `MetricsTable` dataset. Either we can expose it from `MetricsTable` so that we can use it in metrics processor or we can add a check in `HBaseMetricsTableDefinition` with log sampler?